source/applications/advanced/color_balance.py

```diff
@@ -0,0 +1,203 @@
+
+    Returns:
+        mean_red: Red channel mean value
+        mean_green: Green channel mean value
+        mean_blue: Blue channel mean value
+
+    """
+    height = np.shape(rgb)[0]
+    width = np.shape(rgb)[1]
+    red = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        0,
+    ]
+    green = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        1,
+    ]
+    blue = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        2,
+    ]
+    mean_red = np.mean(np.reshape(red, -1, []), dtype=np.float64)
+    mean_green = np.mean(np.reshape(green, -1, []), dtype=np.float64)
+    mean_blue = np.mean(np.reshape(blue, -1, []), dtype=np.float64)
+
+    return mean_red, mean_green, mean_blue
+
+
+def _apply_color_balance(rgb, red_balance, blue_balance):
+    """Apply color balance to RGB image.
+
+    Args:
+        rgb: Input RGB image
+        red_balance: Red balance
+        blue_balance: Blue balance
+
+    Returns:
+        corrected_image: RGB image after color balance
+
+    """
+    default_red_balance = zivid.Settings().red_balance
+    default_blue_balance = zivid.Settings().blue_balance
+    corrected_rgb = np.copy(rgb)
+    corrected_rgb[:, :, 0] = rgb[:, :, 0] * red_balance / default_red_balance
+    corrected_rgb[:, :, 2] = rgb[:, :, 2] * blue_balance / default_blue_balance
+
+    return corrected_rgb
+
+
+def _color_balance_calibration(camera, settings_3d):
+    """Balance color for RGB image by taking images of white surface (piece of paper, wall, etc.) in a loop.
+
+    Args:
+        camera: Zivid camera
+        settings_3d: 3D capture settings
+
+    Returns:
+        corrected_red_balance: Corrected red balance
+        corrected_blue_balance: Corrected blue balance
+
+    """
+    print(f"Starting color balance calibration")
+    corrected_red_balance = 1.0
+    corrected_blue_balance = 1.0
+    settings_list = []
+    settings_list.append(settings_3d)
+    first_iteration = True
+    while True:
+        settings_3d.red_balance = corrected_red_balance
+        settings_3d.blue_balance = corrected_blue_balance
+        frame = zivid.hdr.capture(camera, settings_list)
+        point_cloud = frame.get_point_cloud().to_array()
+        rgb = np.dstack([point_cloud["r"], point_cloud["g"], point_cloud["b"]])
+        if first_iteration:
+            _display_rgb(rgb, "RGB image before color balance (3D capture)")
+            first_iteration = False
+        [mean_red, mean_green, mean_blue] = _compute_mean_rgb(rgb, 100)
+        print(
+            f"Mean color values: R =",
+            int(mean_red),
+            f"G =",
+            int(mean_green),
+            f"B =",
+            int(mean_blue),
+        )
```

Comment:
How about something like this:
```
print(
    (
       "Mean color values: R = "
        f"{int(mean_red)}\n"
        "G = "
        f"{int(mean_green)}\n"
        f"B = "
        f"{int(mean_blue)}\n"
    )
)
```
The f-strings in the current code does not do anything, since you are using comma to separate the arguments of the print

Reply:
Done like this (removed \n and added space so it is in one line):
`        print(
            (
               "Mean color values: R = "
                f"{int(mean_red)} "
                "G = "
                f"{int(mean_green)} "
                f"B = "
                f"{int(mean_blue)} "
            )
        )`